### PR TITLE
python-more-itertools: add new package

### PR DESCRIPTION
**Pull request by ja-pa**
Maintainer: me
Compile tested: Turris Omnia (TOS5), OpenWrt master
Run tested: Turris Omnia (TOS5), OpenWrt master

Description:
This PR is based on #8562 . It splits new packages to individuals PR.

Run tested with simple example:

Test example
```
from more_itertools import flatten
iterable = [(0, 1), (2, 3)]
list(flatten(iterable))
[0, 1, 2, 3]
```

Signed-off-by: Jan Pavlinec <jan.pavlinec@nic.cz>

### CODE REVIEW

File: lang/python/python-more-itertools/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=python-more-itertools
+PKG_VERSION:=7.2.0
+PKG_RELEASE:=1
+
+PYPI_NAME:=more-itertools
+PKG_HASH:=409cd48d4db7052af495b09dec721011634af3753ae1ef92d2b32f73a745f832
+PKG_BUILD_DIR:=$(BUILD_DIR)/more-itertools-$(PKG_VERSION)
```
Comment (jefferyto) [inline lang/python/python-more-itertools/Makefile]: Setting `PKG_BUILD_DIR` shouldn't be necessary.

Comment (ja-pa) [reply lang/python/python-more-itertools/Makefile]: Removed

**Comment by mans0n** [comment]
Please backport this to 19.07 too. `make defconfig` on 19.07 prints dependency warning:
`WARNING: Makefile 'package/feeds/packages/python-zipp/Makefile' has a dependency on 'python3-more-itertools', which does not exist`

**Comment by neheb** [comment]
Done

**Pull request merged**